File: src/main/kotlin/com/nazarethlabs/notes/service/DeleteNoteService.kt
package com.nazarethlabs.notes.service

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.Messages
import com.nazarethlabs.notes.dto.Note
import com.nazarethlabs.notes.helper.DialogHelper
import com.nazarethlabs.notes.helper.FileHelper
import com.nazarethlabs.notes.helper.MessageHelper
import com.nazarethlabs.notes.repository.NoteStorageRepository

class DeleteNoteService {

    fun confirmAndDelete(project: Project, note: Note) {
        val result = DialogHelper.showYesNoDialog(
            project,
            MessageHelper.getMessage("dialog.delete.note.message", note.title),
            MessageHelper.getMessage("dialog.delete.note.title")
        )

        if (result == Messages.YES) {
            FileHelper.deleteFile(note.filePath)

            NoteStorageRepository
                .getInstance()
                .removeNote(project, note.id)
        }
    }
}

File: src/service/DeleteNoteService.ts
import { Note } from '../dto/Note';
import { NoteRepository } from '../repository/NoteRepository';
import { FileHelper } from '../helper/FileHelper';
import { DialogHelper } from '../helper/DialogHelper';

export class DeleteNoteService {
  private readonly repository: NoteRepository;

  constructor(repository: NoteRepository) {
    this.repository = repository;
  }

  async confirmAndDelete(notes: Note[]): Promise<boolean> {
    const names = notes.slice(0, 5).map((n) => `"${n.title}"`);
    const extra = notes.length > 5 ? ` and ${notes.length - 5} more` : '';
    const confirmed = await DialogHelper.showConfirmation(
      `Delete ${names.join(', ')}${extra}? This action cannot be undone.`,
    );

    if (!confirmed) {
      return false;
    }

    for (const note of notes) {
      FileHelper.deleteFile(note.filePath);
      this.repository.removeNote(note.id);
    }

    return true;
  }
}
